Connection/Connection.php

```diff
@@ -337,14 +337,31 @@ public function disconnect()
 
     /**
      * Set socket.
+     * A wrapper can be registered to handle application layer transport
      *
-     * @param   string  $socket    Socket URI.
+     * @param   string  $socketUri    Socket URI.
      * @return  \Hoa\Socket
+     * @throws  \Hoa\Socket\Exception
      */
-    protected function setSocket($socket)
+    protected function setSocket($socketUri)
     {
+        $parsed = parse_url($socketUri);
+        if (null !== $wrapper =
+            Socket\Transport::getWrapper($parsed['scheme'])) {
```

Comment:
On the same line (do not break after `=`).
You must test that the `scheme` key exists. It must exists but let `Hoa\Socket\Socket` handles this error for you.